src/Npgsql/NpgsqlConnector.cs

```diff
@@ -546,15 +546,6 @@ string GetUsername()
             if (username?.Length > 0)
                 return username;
 
-#if NET461
-            if (PGUtil.IsWindows && Type.GetType("Mono.Runtime") == null)
-            {
-                username = WindowsUsernameProvider.GetUsername(Settings.IncludeRealm);
-                if (username?.Length > 0)
-                    return username;
-            }
-#endif
-
```

Comment:
This looks a bit more invasive than it needs to be.

I think we could at least try to solve this using [`Environment.UserName`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.username?view=netcore-3.1#System_Environment_UserName)

Reply:
We consult Environment.UserName a few lines below, right? Or am I missing something?

There's probably some stuff we can do better for username detection - the Kerberos logic is currently for non-Windows only, though I think Windows also has klist and may work... But let's leave that out of scope, and also wait for people to ask for it first (I hardly remember anyone doing that).

Reply:
> We consult Environment.UserName a few lines below, right? Or am I missing something?

I missed that part because I was only looking at the diff.
In that case I don't see any problem.